pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3246,16 +3247,18 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
 
     // Check that all the segments from 'segmentsFrom' exist in the table
     Set<String> segmentsForTable = new HashSet<>(getSegmentsFor(tableNameWithType, true));
-    Preconditions.checkArgument(segmentsForTable.containsAll(segmentsFrom), String.format(
-        "Not all segments from 'segmentsFrom' are available in the table. (tableName = '%s', segmentsFrom = '%s', "
-            + "segmentsTo = '%s', segmentsFromTable = '%s')", tableNameWithType, segmentsFrom, segmentsTo,
-        segmentsForTable));
+    Set<String> unavailableSegmentsInFrom = Sets.difference(new HashSet<>(segmentsFrom), segmentsForTable);
+    Preconditions.checkArgument(unavailableSegmentsInFrom.isEmpty(), String.format(
+        "'%s' from 'segmentsFrom' are unavailable in the table. (tableName = '%s', segmentsFrom = '%s', "
+            + "segmentsTo = '%s', segmentsFromTable = '%s')", unavailableSegmentsInFrom, tableNameWithType,
```

Comment:
In this case, we can actually remove `segmentsFromTable`? 

In many cases this is not too useful. Also, this list can be very long in general (e.g. can be 10k segments)

Reply:
good point, let me remove it